torch/_dynamo/variables/builder.py

```diff
@@ -2211,8 +2225,36 @@ def _automatic_dynamic(
                             e.size(i),
                             dim,
                         )
+                        has_size_changed = True
```

Comment:
This looks wrong. If size has changed in a previous iteration, you need this boolean to still be True (in this case, dim is None).